Custom colors first pass

celloexpressions opened the pull request:
See #116.

celloexpressions committed:
Merge pull request #1 from WordPress/master

Update from upstream

celloexpressions committed:
Add dark stylesheet

celloexpressions committed:
Add postMessage support for color schemes

celloexpressions committed:
Enqueue the dark colorscheme when needed

celloexpressions committed:
Introduce color patterns file for custom colors

celloexpressions committed:
Add color schemes to the customizer

celloexpressions committed:
Unify base colors

celloexpressions committed:
Add missing comma

celloexpressions committed:
Add additional transparent button background overrides

celloexpressions committed:
Update dark colorscheme colors per @melchoyce

celloexpressions committed:
Add a visual notice that the hue slider is temporary

ntwb commented on assets/css/colors-dark.css:
```diff
@@ -0,0 +1,464 @@
+	/* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#000000+0,000000+100&0+0,0.3+100 */
+	background: -moz-linear-gradient(top,  rgba(255,255,255,0) 0%, rgba(255,255,255,0.3) 100%); /* FF3.6-15 */
+	background: -webkit-linear-gradient(top,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 100%); /* Chrome10-25,Safari5.1-6 */
+	background: linear-gradient(to bottom,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
```
Expected single space after commas on the above 3 lines

celloexpressions replied:
This is a direct copy/paste from style.css, but swapping out the colors; it will need to be updated there as well.

ntwb replied:
Yup, #169 gets all the coding standards sorted for style.css

ntwb commented on assets/css/colors-dark.css:
```diff
@@ -0,0 +1,464 @@
+		/* Permalink - use to edit and share this gradient: http://colorzilla.com/gradient-editor/#000000+0,000000+100&0+0,0.3+75 */
+		background: -moz-linear-gradient(top,  rgba(255,255,255,0) 0%, rgba(255,255,255,0.3) 75%, rgba(255,255,255,0.3) 100%); /* FF3.6-15 */
+		background: -webkit-linear-gradient(top,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 75%,rgba(255,255,255,0.3) 100%); /* Chrome10-25,Safari5.1-6 */
+		background: linear-gradient(to bottom,  rgba(255,255,255,0) 0%,rgba(255,255,255,0.3) 75%,rgba(255,255,255,0.3) 100%); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
```
Expected single space after commas on the above 3 lines

ntwb reviewed (changes requested):
Added a couple of comments for spaces after commas

celloexpressions committed:
Add spaces in rgba() colors

Note that this was copied from style.css.

ryelle commented on inc/customizer.php:
```diff
@@ -142,6 +183,19 @@ function twentyseventeen_sanitize_layout( $input ) {
+function twentyseventeen_sanitize_colorscheme( $input ) {
+	$valid = array( 'light', 'dark', 'custom' );
+
+	if ( array_key_exists( $input, $valid ) ) {
```
This should be `in_array`, since `$valid` is not a keyed array. Right now this doesn't save `colorscheme`.

celloexpressions committed:
Fix JSHint errors

celloexpressions committed:
Fix sanitization, add link to core hue ticket

celloexpressions committed:
Add package and since headers

celloexpressions committed:
Update inline docs

celloexpressions committed:
Add missing space

celloexpressions commented:
The one remaining Travis CI failure is for line endings, which I'm unable to change from here. Someone will need to fix that on or after merge.

This should be ready to go. After the initial merge, someone will need to go through and merge all changes to `style.css` after October 4th that impact colors in any way back to `color-patterns.php` and `colors-dark.css`. All future commits to `style.css` that touch colors need to consider the color files as well now too. This applies to any color, including white or black.

celloexpressions committed:
Dark scheme: don't invert gradient over headers

celloexpressions committed:
Dark scheme: lighten colors

Per @melchoyce, switching #999 to #bbb. Assuming #aaa will need to go to #ccc in the process so that it doesn't become darker than the other pattern.

celloexpressions committed:
Dark scheme: add dark background for inputs

The base colorscheme doesn't specify this, so it wasn't being inverted but the text color was. Selectors copied from where the text color is inverted.

celloexpressions committed:
Adjust saturation

Adjust saturation to 50% per @melchoyce, and reduce the saturation on body text (base: #333) by 20% of the base saturation (should come out to 40%). Needs testing.
